### PR TITLE
Update Sarevok: remove on-hit melee effect and patch in dynamic XP ad…

### PR DESCRIPTION
**Pull request by critto-bg**
…justment upon joining the party

### CODE REVIEW

File: ImprovedAnvil/lib/creatures/04_patches_joinable.tph [hunk 1]
```diff
@@ -1535,6 +1541,7 @@ ACTION_FOR_EACH ~join_dialogue~ IN
   ~viconi~
   ~yoshimo~
   ~hexxat~
+  ~sarev25a~
```
Comment (bulian-bg) [inline ImprovedAnvil/lib/creatures/04_patches_joinable.tph]: I like the dynamic dialogue approach but Sarevok is somewhat unique in that his stats allow dual classing to either thief or mage, so it seems like this approach locks him into a fighter, whereas the script I laid out allowed the player to dualclass in the pocket plane if desired before moving on with the game.

Comment (roman-d) [reply ImprovedAnvil/lib/creatures/04_patches_joinable.tph]: We could try to come up with something that postpones running his XP script until the party enters Saradush, for example. Would that work? And mention the thing in the readme for players to be aware about this.

Comment (bulian-bg) [reply ImprovedAnvil/lib/creatures/04_patches_joinable.tph]: I'm sure there is a way to do that, but I did a quick test using Jan and it seems the XP gain comes from the second class, so theoretically Sarevok could end up with 10.7 million XP at the start of TOB.  

The simplest option that preserves dual classing seems to be the way I implemented it, possibly with a couple of extra conditions based on PC XP to hone in on a target XP, but that does make it a one off compared to other NPCs..  It also seems reasonable that Sarevok's XP could be lower than the PCs XP and it wouldn't matter much at that stage of the game.

I'll leave it up to you

Comment (critto-bg) [reply ImprovedAnvil/lib/creatures/04_patches_joinable.tph]: Well, I haven't seen game in action for so long I can only assume that you're right. I was just trying to streamline the process for everyone. With my method he will definitely not have more XP than any other character in the party. Or are you worried about him being the same level as the other while multi-classed?

Would it upset the balance much? From the lore stand-point I don't see much problem, after it's Sarevok we're talking about, he's an extraordinary individual.

**Comment by critto-bg** [comment]
@bulian-bg I've updated this to your version of the patch for now, although I've re-implemented it a bit as an area's script extension.

**Review by bulian-bg — APPROVED** [review]
Checks out!  Nice!

**Comment by critto-bg** [comment]
Cool. Rebase, merged and pushed into master.

**Pull request closed without merging**